trieste/acquisition/function.py

```diff
@@ -785,13 +785,14 @@ def prepare_acquisition_function(
 
         _pf = Pareto(mean)
         _reference_pt = get_reference_point(_pf.front)
-        return expected_hv_improvement(model, _pf, _reference_pt)
+        _partition_bounds = prepare_default_non_dominated_partition_bounds(
+            _pf.front, tf.constant([-1e10] * mean.shape[-1], dtype=_pf.front.dtype), _reference_pt
```

Comment:
why do we have -1e10 hardcoded into this?

Reply:
Ideally it shall just be `-inf`, but i recall very earlier this (using inf in bounds) will impose some `NaN` issue on gradient calculation of acq function in optimization, so this hard code is for improving the numerical stability. Not idea tbh, but am not sure if there are some better way of handling this atm.

Reply:
yep makes sense. Maybe add this as a comment in the code